crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1330,6 +1330,136 @@ pub mod lifetimeless {
     pub type SCommands = crate::system::Commands<'static, 'static>;
 }
 
+/// A helper for using system parameters in generic contexts
+///
+/// This type is a [`SystemParam`] adapter which always has
+/// `Self::Fetch::Item == Self` (ignoring lifetimes for brevity),
+/// no matter the argument [`SystemParam`] (`P`) (other than
+/// that `P` must be `'static`)
```

Comment:
This restriction is unfortunate, but pretty fundamental as far as I can tell. 
This is because we need be able to refer to `P` in `SystemParamState`, which reasonably needs to be `'static`